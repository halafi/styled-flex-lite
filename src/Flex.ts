import styled from 'styled-components';

type Props = {
  as?: string; // polymorphic prop
  flexDirection?: 'row' | 'column' | 'row-reverse' | 'column-reverse';
  alignItems?: 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
  justifyContent?:
    | 'flex-start'
    | 'flex-end'
    | 'center'
    | 'space-between'
    | 'space-around'
    | 'space-evenly';
  textAlign?: 'left' | 'right' | 'center' | 'justify' | 'inherit';
  flexWrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
};

const Flex = styled.div<Props>`
  display: flex;
  ${({ flexDirection }) => flexDirection && `flex-direction: ${flexDirection};`};
  ${({ alignItems }) => alignItems && `align-items: ${alignItems};`};
  ${({ justifyContent }) => justifyContent && `justify-content: ${justifyContent};`};
  ${({ textAlign }) => textAlign && `text-align: ${textAlign};`};
  ${({ flexWrap }) => flexWrap && `flex-wrap: ${flexWrap};`};
  box-sizing: border-box;
`;

export default Flex;
