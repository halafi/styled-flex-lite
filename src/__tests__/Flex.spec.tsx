import React from 'react';
import styled from 'styled-components';
import * as renderer from 'react-test-renderer';
import Flex from '..';
import 'jest-styled-components';

const StyledFlex = styled(Flex)`
  padding: 4px;
`;

describe('#Flex', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StyledFlex flexDirection="column" justifyContent="flex-end">
          <Flex>1</Flex>
          <Flex>2</Flex>
        </StyledFlex>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
