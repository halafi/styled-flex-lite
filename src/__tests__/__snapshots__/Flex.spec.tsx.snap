// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Flex renders correctly 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    1
  </div>
  <div
    className=""
  >
    2
  </div>
</div>
`;
